bq-connector/docai_bq_connector/exception/InvalidGcsUriError.py

```diff
@@ -0,0 +1,22 @@
+#
+# Copyright 2022 Google LLC
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+class InvalidGcsUriError(Exception):
```

Comment:
Is this exception really needed? won't the Google Cloud APIs throw a `NOT FOUND` or `INVALID ARGUMENT` exception for this?

Reply:
Not exactly, the following logic uses the parsed bucket to get bucket - that would throw an exception if the bucket didn't exist, but a prefix missing for the bucket search wouldn't cause any error.

Also, I added this exception to resolve the issue raised by the linter. The linter didn't like that the Optional type response from the re.match wasn't checked. I can eliminate this exception and just default the bucket and prefix vars to empty strings too.